plugins/outputs/groundwork/log_adapter.go

```diff
@@ -0,0 +1,84 @@
+package groundwork
+
+import (
+	"context"
+	"encoding/json"
+	"log/slog"
+	"strings"
+
+	"github.com/influxdata/telegraf"
+)
+
+// NewLogger creates telegraf.Logger adapter for slog.Logger
+func NewLogger(l telegraf.Logger) *slog.Logger {
+	return slog.New(&TlgHandler{Log: l})
+}
+
+// TlgHandler translates slog.Record into telegraf.Logger call
+// inspired by https://github.com/golang/example/blob/master/slog-handler-guide/README.md
+type TlgHandler struct {
+	attrs  []slog.Attr
+	groups []string
+
+	Log telegraf.Logger
+}
+
+// Enabled implements slog.Handler interface
+// It interprets errors as errors and everything else as debug.
+func (h *TlgHandler) Enabled(_ context.Context, level slog.Level) bool {
+	if level == slog.LevelError {
+		return h.Log.Level() >= telegraf.Error
+	}
+	return h.Log.Level() >= telegraf.Debug
+}
+
+// Handle implements slog.Handler interface
+// It interprets errors as errors and everything else as debug.
+func (h *TlgHandler) Handle(_ context.Context, r slog.Record) error {
+	attrs := make([]slog.Attr, 0, 2+len(h.attrs)+r.NumAttrs())
+	attrs = append(attrs,
+		slog.String("logger", strings.Join(h.groups, ",")),
+		slog.String("message", r.Message),
+	)
+	for _, attr := range h.attrs {
+		if v, ok := attr.Value.Any().(json.RawMessage); ok {
+			attrs = append(attrs, slog.String(attr.Key, string(v)))
+			continue
+		}
+		attrs = append(attrs, attr)
+	}
```

Comment:
Couldn't this conversion be already done in `WithAttrs`? This would save a few cycles and simplify code.

Reply:
done